pgmpy/tests/test_base/test_DAG.py

```diff
@@ -678,6 +681,346 @@ def test_edge_strength_skip_latent_edges(self):
+
+        # Should raise ValueError when edge strengths are missing
+        with self.assertRaises(ValueError):
+            dag_no_strength.to_daft(
+                node_pos={"X": (0, 0), "Y": (1, 0)}, plot_edge_strength=True
+            )
+
+        with self.assertRaises(ValueError):
+            dag_no_strength.to_graphviz(plot_edge_strength=True)
+
+    def test_edge_strength_individual_errors(self):
+        """Test individual edge errors in edge strength plotting"""
+        dag = DAG([("A", "B"), ("C", "B")])
+
+        # Set strength for only one edge
+        dag.edges[("A", "B")]["strength"] = 0.123
+        # Leave ("C", "B") without strength
+
+        # Test to_daft method with individual edge error
+        with self.assertRaises(ValueError):
+            dag.to_daft(
+                node_pos={"A": (0, 0), "B": (1, 0), "C": (0, 1)},
+                plot_edge_strength=True,
+            )
+
+        # Test to_graphviz method with individual edge error
+        with self.assertRaises(ValueError):
+            dag.to_graphviz(plot_edge_strength=True)
+
+    def test_edge_strength_comprehensive_errors(self):
+        """Test comprehensive edge strength error scenarios"""
+        # Test DAG with no edge strengths at all
+        dag_no_strengths = DAG([("X", "Y"), ("Z", "Y")])
+
+        # Test to_daft with no strengths - should raise ValueError for all edges
+        with self.assertRaises(ValueError):
+            dag_no_strengths.to_daft(
+                node_pos={"X": (0, 0), "Y": (1, 0), "Z": (0, 1)},
+                plot_edge_strength=True,
+            )
+
+        # Test to_graphviz with no strengths - should raise ValueError for all edges
+        with self.assertRaises(ValueError):
+            dag_no_strengths.to_graphviz(plot_edge_strength=True)
+
+    def test_edge_strength_plotting_missing_individual(self):
+        """Test edge strength plotting when individual edges are missing strengths"""
+        # Test scenario for to_daft method missing individual edge strengths
+        dag = DAG([("A", "B"), ("C", "B"), ("D", "E")])
+
+        # Only set strength for some edges to trigger individual errors
+        dag.edges[("A", "B")]["strength"] = 0.456
+        dag.edges[("D", "E")]["strength"] = 0.789
+        # Leave ("C", "B") without strength to trigger the error
+
+        # Test to_daft method - this should raise ValueError
+        with self.assertRaises(ValueError):
+            dag.to_daft(
+                node_pos={
+                    "A": (0, 0),
+                    "B": (1, 0),
+                    "C": (0, 1),
+                    "D": (2, 0),
+                    "E": (3, 0),
+                },
+                plot_edge_strength=True,
+            )
+
+        # Test to_graphviz method - this should raise ValueError
+        with self.assertRaises(ValueError):
+            dag.to_graphviz(plot_edge_strength=True)
+
+    def test_edge_strength_plotting_format_precision(self):
```

Comment:
Don't really need this. This is a lot of tests for a relatively simple functionality. This would make maintenance harder in the long term.